ethereum/statetransition/src/main/java/tech/pegasys/teku/statetransition/forkchoice/ForkChoiceBlobSidecarsAvailabilityChecker.java

```diff
@@ -53,333 +33,93 @@
+    final SignedBeaconBlock block = blockBlobSidecarsTracker.getBlock().orElseThrow();
 
-    try {
-      miscHelpers.validateBlobSidecarsBatchAgainstBlock(
-          blobSidecars, block, kzgCommitmentsFromBlockSupplier.get());
-
-      if (!miscHelpers.verifyBlobKzgProofBatch(kzg, blobSidecars)) {
-        return BlobSidecarsAndValidationResult.invalidResult(blobSidecars);
+    for (final BlobSidecar blobSidecar : blobSidecars) {
+      if (!blobSidecar.isKzgAndInclusionProofValidated()) {
+        return BlobSidecarsAndValidationResult.notAvailable(
+            new IllegalStateException("Blob sidecars are not validated"));
       }
-    } catch (final Exception ex) {
-      return BlobSidecarsAndValidationResult.invalidResult(blobSidecars, ex);
-    }
-
-    return BlobSidecarsAndValidationResult.validResult(blobSidecars);
-  }
-
-  /**
-   * Step 1 of 2
-   *
-   * <p>This function performs a partial validation with blobs that are currently available from the
-   * tracker. If all blobs are already available, it performs a full validation.
-   *
-   * @return a validation result only in case it is a definitive result. If the validation needs to
-   *     be completed it returns empty
-   */
-  private Optional<BlobSidecarsAndValidationResult> validateImmediatelyAvailable() {
-    final List<KZGCommitment> kzgCommitmentsInBlock = kzgCommitmentsFromBlockSupplier.get();
-
-    final List<BlobSidecar> blobSidecarsToValidate;
-
-    final boolean performCompleteValidation = blockBlobSidecarsTracker.isCompleted();
-
-    if (performCompleteValidation) {
-      // the tracker contains all required blobs for our block: we can perform a complete validation
-      blobSidecarsToValidate = List.copyOf(blockBlobSidecarsTracker.getBlobSidecars().values());
-      LOG.debug(
-          "The expected {} BlobSidecars have already been received. Performing full validation.",
-          blobSidecarsToValidate.size());
-    } else {
-      // prepare partial validation by matching the currently available blobs with the corresponding
-      // commitments from the block
-      blobSidecarsToValidate =
-          blockBlobSidecarsTracker.getBlobSidecars().values().stream().toList();
-
-      LOG.debug(
-          "{} out of {} BlobSidecars have been received so far. Performing partial validation.",
-          blobSidecarsToValidate.size(),
-          kzgCommitmentsInBlock.size());
-
-      if (blobSidecarsToValidate.isEmpty()
-          && !kzgCommitmentsInBlock.isEmpty()
-          && isBlockOutsideDataAvailabilityWindow()) {
-        // there are no available blobs so far, but we are outside the availability window. We can
-        // skip additional checks
-        return Optional.of(BlobSidecarsAndValidationResult.NOT_REQUIRED);
+      if (!blobSidecar.isSignatureValidated()
+          && !blobSidecar
+              .getSignedBeaconBlockHeader()
+              .hashTreeRoot()
+              .equals(block.hashTreeRoot())) {
```

Comment:
> So if we end up processing blobs coming from RPC source, we are not checking the block header in the blobSidecar (correct me if i'm wrong :) )

Ok I see what you mean. One should keep in mind the whole flow to understand that...